plugin/pkg/admission/runtimeclass/admission.go

```diff
@@ -228,7 +228,7 @@ func validateOverhead(a admission.Attributes, pod *api.Pod, runtimeClass *nodev1
 		}
 	} else {
 		// If RuntimeClass with Overhead is not defined but an Overhead is set for pod, reject the pod
-		if pod.Spec.Overhead != nil {
+		if pod.Spec.Overhead != nil && len(pod.Spec.Overhead) > 0 {
```

Comment:
we can just use `len(pod.Spec.Overhead) > 0`: https://go.dev/ref/spec#Length_and_capacity